### PR TITLE
haskellPackages.minio-hs: Use crypton-connection instead of unmaintained connection

### PR DESCRIPTION
**Pull request by mpscholten**
Fixes build of `minio-hs` by applying the patch of https://github.com/minio/minio-hs/pull/191

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -2409,11 +2409,19 @@ self: super: {
...
   # https://github.com/minio/minio-hs/issues/165
+  # https://github.com/minio/minio-hs/pull/191 Use crypton-connection instead of unmaintained connection
   minio-hs = overrideCabal (drv: {
     testFlags = [
       "-p" "!/Test mkSelectRequest/"
     ] ++ drv.testFlags or [];
-  }) super.minio-hs;
+    patches = drv.patches or [ ] ++ [
+      (pkgs.fetchpatch {
+        name = "use-crypoton-connection.patch";
```
Comment (ncfavier) [inline pkgs/development/haskell-modules/configuration-common.nix:2419]:
```suggestion
        name = "use-crypton-connection.patch";
```

**Comment by ncfavier** [comment]
The package is still marked transitively broken because of `connection`. Could you run `./maintainers/scripts/haskell/regenerate-transitive-broken-packages.sh` to unmark it as broken?

**Comment by ncfavier** [comment]
Er actually you need to run `./maintainers/scripts/haskell/regenerate-hackage-packages.sh`, sorry

**Comment by mpscholten** [comment]
Thanks, just tried it but got this error:
```
marc@Marcs-M3-Max nixpkgs % ./maintainers/scripts/haskell/regenerate-hackage-packages.sh
Obtaining Hackage data …
Generating compiler configuration …
Running hackage2nix to regenerate pkgs/development/haskell-modules/hackage-packages.nix …
hackage2nix: constraint doctest >=0.18 && <0.19 cannot be resolved in Hackage
This could be due the package being missing in the hackage directory
or the file system not being case sensitive.

CallStack (from HasCallStack):
  error, called at hackage2nix/Main.hs:205:34 in main:Main
hackage2nix: thread blocked indefinitely in an MVar operation
```

Have you seen this before?

**Comment by mpscholten** [comment]
Just manually removed it from the transitive broken list, hope this works as well 👍

**Pull request merged**